server/oauth.go

```diff
@@ -56,8 +56,7 @@ func (s *Server) registerOAuthRoutes(g *echo.Group) {
 				Code:         req.Code,
 			}
 		}
-
-		oauthExchange.RedirectURL = fmt.Sprintf("%s:%d/oauth/callback", s.profile.FrontendHost, s.profile.FrontendPort)
+		oauthExchange.RedirectURL = req.RedirectURL
```

Comment:
We have other places suffering this. e.g. in the same file at line 99.

If we pass it from frontend, then let's fix them together